### PR TITLE
[Merged by Bors] - feat: Alexandrov-discrete spaces

### PR DESCRIPTION
**Pull request by YaelDillies**
We define Alexandrov-discrete spaces as topological spaces where the intersection of a family of open sets is open.

This PR only gives a minimal API because the goal is to ensure that lemma names like `isOpen_sInter` are free to use for `AlexandrovDiscrete`. The existing lemmas are getting prefixed by `Set.Finite` or suffixed by `_of_finite`.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Topology/AlexandrovDiscrete.lean [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+Copyright (c) 2023 Yaël Dillies. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yaël Dillies
+-/
+import Mathlib.Topology.Bases
+import Mathlib.Topology.Inseparable
+import Mathlib.Topology.SubsetProperties
+
+/-!
+# Alexandroff-discrete topological spaces
```
Comment (alexjbest) [inline Mathlib/Topology/AlexandrovDiscrete.lean:11]:
```suggestion
# Alexandrov-discrete topological spaces
```
Maybe just put the ff version in the tags, and stay consistent with one choice for the rest of the file

Comment (YaelDillies) [reply Mathlib/Topology/AlexandrovDiscrete.lean]: I tried to be consistent... and failed 😅

**Comment by alexjbest** [comment]
LGTM

maintainer merge

**Comment by riccardobrasca** [comment]
Thanks!

bors merge

**Comment by riccardobrasca** [comment]
bors d+

**Comment by YaelDillies** [comment]
Not sure it'll go through but the queue is empty, so let's try.

bors merge

**Pull request closed without merging**